### PR TITLE
Test scikit-build-core tiledb-py

### PR DESCRIPTION
**Pull request by dudoslav**
This PR is purely for testing purpose in order to see if this PR will work:
https://github.com/TileDB-Inc/TileDB-Py/pull/1988

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Review by jdblischak — CHANGES_REQUESTED** [review]
You need to add `scikit-build-core` to the host requirements

**Comment by dudoslav** [comment]
And cmake

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,11 +7,12 @@ package:
...
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  git_rev: db/scikit-build-core
+  git_url: https://github.com/TileDB-Inc/TileDB-Py.git
 
 build:
+  script: TILEDB_PATH=${PREFIX} {{ PYTHON }} -m pip install --no-build-isolation --no-deps --ignore-installed -v . # [not win]
+  script: set "TILEDB_PATH=${PREFIX}" && {{ PYTHON }} -m pip install --no-build-isolation --no-deps --ignore-installed -v . # [win]
```
Comment (jdblischak) [inline recipe/meta.yaml:15]:
```suggestion
  script: set "TILEDB_PATH=${PREFIX}" && {{ PYTHON }} -m pip install --no-build-isolation --no-deps --ignore-installed -v .  # [win]
```

Not important for this test PR, but FYI this is what the linter is complaining about. It wants two spaces between the end of the line and the start of the jinja2 selector

**Review by jdblischak — APPROVED** [review]
Looks good! I confirmed that `libtiledb.so` isn't getting copied into the Python installation directory

```
  *** Installing project into wheel...
  -- Install configuration: "RelWithDebInfo"
  -- Installing: /tmp/tmphwnd407a/wheel/platlib/tiledb/main.cpython-39-x86_64-linux-gnu.so
  -- Set non-toolchain portion of runtime path of "/tmp/tmphwnd407a/wheel/platlib/tiledb/main.cpython-39-x86_64-linux-gnu.so" to "/home/conda/feedstock_root/build_artifacts/tiledb-py_1720685810279/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib"
  -- Installing: /tmp/tmphwnd407a/wheel/platlib/tiledb/libtiledb.cpython-39-x86_64-linux-gnu.so
  -- Set non-toolchain portion of runtime path of "/tmp/tmphwnd407a/wheel/platlib/tiledb/libtiledb.cpython-39-x86_64-linux-gnu.so" to "/home/conda/feedstock_root/build_artifacts/tiledb-py_1720685810279/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib"
  -- Installing: /tmp/tmphwnd407a/wheel/platlib/tiledb/cc.cpython-39-x86_64-linux-gnu.so
  -- Set non-toolchain portion of runtime path of "/tmp/tmphwnd407a/wheel/platlib/tiledb/cc.cpython-39-x86_64-linux-gnu.so" to "/home/conda/feedstock_root/build_artifacts/tiledb-py_1720685810279/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib"
  *** Making wheel...
  *** Created tiledb-0.30.3.dev109+gd18362b1-cp39-cp39-linux_x86_64.whl...
  Building wheel for tiledb (pyproject.toml): finished with status 'done'
  Created wheel for tiledb: filename=tiledb-0.30.3.dev109+gd18362b1-cp39-cp39-linux_x86_64.whl size=31466445 sha256=8e0d9e2bb2e98f24384f25f562cf5b13f6a9e2a5c302dc34eefe4334b3dd72d2
  Stored in directory: /tmp/pip-ephem-wheel-cache-tbx35kw5/wheels/ae/cf/aa/c54d1dab71887bebd6619bd97ce2209e7935ce2ffdc4a8d52b
Successfully built tiledb
Installing collected packages: tiledb
Successfully installed tiledb-0.30.3.dev109+gd18362b1
```

File: recipe/meta.yaml (hunk 1 above)
Comment (jdblischak) [inline recipe/meta.yaml:15]:
```suggestion
  script: set "TILEDB_PATH=%LIBRARY_PREFIX%" && {{ PYTHON }} -m pip install --no-build-isolation --no-deps --ignore-installed -v . # [win]
```

The env var on Windows is `%LIBRARY_PREFIX%`. This is what was used previously in this feedstock (see above), and is what is used in [tiledb-feedstock](https://github.com/conda-forge/tiledb-feedstock/blob/f7f5e16ba8630a7fd074ef1a8b456b3a28935172/recipe/bld.bat#L13)

**Pull request closed without merging**